File: Prueba Corte 1 - Juan Andrés Villabona/Evaluacion Corte 1/src/evaluacion/corte/pkg1/Figura.java
package evaluacion.corte.pkg1;


public abstract class Figura {
    public String nomFigura;
    public double area;
    public double perim;
    
    public void establecerNomFigura(String nombre){
        this.nomFigura = nombre;
    }
    
    public String obtenerNomFigura(){
        return this.nomFigura;
    }
    
    // Métodos abstractos
    public abstract void calcularArea();
    public abstract void calcularPerim();
    
    public double obtenerArea(){
        return this.area;
    }
    
    public double obtenerPerim(){
        return this.perim;
    }
    
}

File: G800-DavidCifuentesVilla/CORTE1/Corte1Prueba/src/app/clases/figura.ts
export abstract class Figura {

    private nomFigura: string;
    private area: number;
    private perim: number;

    establecerNomFigura(nombre: string){
        this.nomFigura = nombre;
    }

    obtenerNomFigura(): string{
        return this.nomFigura;
    }

    abstract calcularArea();
    abstract calcularPerimetro();

    obtenerArea(): number{
        return this.area;
    }

    obtenerPerimetro(): number{
        return this.perim;
    }
}
